### PR TITLE
Improve Read Speed

### LINKED ISSUE #267: Reading From .ultra Too Slow
The function `pydvpy.read()` is very slow. It takes over a second to parse a 200,000 line file.

I believe there is room for improvement. Profiling shows that a significant amount of time is used in `len`, `str.strip`, `str.split`, `str.lower`, and `list.append`. 

There is also a potential resource leak if an exception is raised during parsing. The file `f` will not be closed. This can be fixed by using a context manager. (`with open(fname, 'r') as f:`)

### PR DESCRIPTION
**Pull request by fillmore1**
### Description

Resolves #267 

Update `pydvpy.read()` to improve performance

### Checklist:

- [x] I have run the image tests on RZ and they pass.
- [ ] I have run the WSC tests on RZ and they pass.
- [x] I have build the docs on sphinx and they look correct.
- [x] I have updated the release notes.
- [x] I have run `pydv/scripts/update_version_and_date.py <major, minor, patch>`.

### CODE REVIEW

**Review by rusu24edward — CHANGES_REQUESTED** [review]
Thanks for the PR! Two things that I listed above. Unfortunately the new reader is unable to pick up some of the nuances of what defines a new curve. In some tests, it adds too many curves, in other tests it does not add enough. I can send you the test files on RZ if you want to iterate faster.

File: pydv/pydvpy.py [hunk 1]
```diff
@@ -437,146 +437,84 @@ def read(fname, gnu=False, xcol=0, verbose=False, pattern=None, matches=None):
...
+        build_list_y = list()
+        current = None
+        with open(file_name, 'r') as f:
+            split_lines = map(str.split, map(str.strip, f))
+
+            for split_line in split_lines:
+                if split_line[0] in {'##', 'end', 'End', 'END'}:
+                    continue
+
+                elif split_line[0] == '#':  # check for start of new curve
```
[the remark this answers is not in the document]
Comment (fillmore1) [reply pydv/pydvpy.py:481]: It seems like `##` should be used for that kind of comment, but I understand keeping it for backwards compatibility. The new commits should resolve this.

File: pydv/pydvpy.py [hunk 2]
```diff
@@ -437,146 +437,84 @@ def read(fname, gnu=False, xcol=0, verbose=False, pattern=None, matches=None):
...
+
+                    if current:
+                        curve_list.append(bundle_curve(current, build_list_x, build_list_y))
+                        build_list_x = list()
+                        build_list_y = list()
+
+                    curve_name = ' '.join(split_line)
```
[the remark this answers is not in the document]
Comment (fillmore1) [reply pydv/pydvpy.py]: I forgot to exclude the first token. My bad.
The new commits should resolve this.

File: pydv/pydvpy.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rusu24edward) [reply pydv/pydvpy.py:481]: I agree, but years of users using "#" for generic comments outweighs me ;)

**Review by rusu24edward — CHANGES_REQUESTED** [review]
The issue with the title having "#" has been resolved. And it appears that you are not storing more curves than actually exists in the ultra file. However, the new reader still struggles to stores curves for many of the test ultra files.

**Review by rusu24edward — CHANGES_REQUESTED** [review]

**Comment by rusu24edward** [comment]
Closing in favor of #271

**Pull request closed without merging**